src/components/HeatmapAndTable.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import Heatmap from './Heatmap';
+import PostTable from './PostTable';
+
+const comparePosts = (post1, post2) => {
+  const date1 = new Date(post1.data.created_utc * 1000);
+  const date2 = new Date(post2.data.created_utc * 1000);
+  const comparison = date1.getMinutes() - date2.getMinutes();
+  if (comparison === 0) {
+    return date1.getSeconds() - date2.getSeconds();
+  }
+  return comparison;
+};
+
+const getPostsAndCountByHour = (posts) => {
+  const countsByHour = Array(7)
+    .fill()
+    .map(() => Array(24).fill(0));
+
+  const postsByHour = {};
```

Comment:
I had that setup before, but weirdly when I pushed to posts[day][hour] it would push the posts to posts[day] instead of the hour array.  So I moved to the object instead.  Was I using the 2d array notation incorrectly?

my push code was `postsByHour[day][hour].push(post)`

Reply:
No, that looks fine. No idea what could have gone wrong there. If you still have the complete code I might be able to tell. Otherwise I'd just try it again. Sometimes problems solve themselves this way :smile:

Reply:
I changed it back (left the other code commented out) and it reverted back to adding it to the `postsByHour[day]' 😆  Thoughts on why it would do that?

Reply:
Not sure... Maybe it's just the view here on GitHub? I found the new code in the [useFetchPosts hook](https://github.com/ooloo-io/reddit-timer-estuelke/blob/5438fea86a07af355944abc2b2c9f9fa6c37bb00/src/hooks/useFetchPosts.js#L70). That looks like what you described (commented out code and so on)

Reply:
Okay, after some sleuthing and testing, I figured out that using

```javascript
Array(7)
    .fill()
    .map(() => Array(24).fill([]));
```
causes the `Array(24)` to be filled with all the same `[]` object and thus they all get updated (see https://stackoverflow.com/a/34435201/13175050).

I thought for a moment and realized that the `Array(7)` portion isn't getting filled with the same object because of `map`.  So I ended up with:

```javascript
const postsByHour = Array(7)
    .fill()
    .map(() => Array(24).fill().map(() => []));
```

And now it works! Yay! 🎉🎉🎉🎉

Reply:
Oh yeah, that makes sense. The same array is used all the time! Oh man, that's really not obvious. I can imagine you spent some time on it. Congrats :tada:

Reply:
Yeah it took me awhile to figure that out.  But now I will know if I run into this in the future!